Libraries/CO.CDP.Mvc.Validation/EmailAddressValidator.cs

```diff
@@ -0,0 +1,92 @@
+using System.Globalization;
+using System.Text.RegularExpressions;
+
+namespace CO.CDP.Mvc.Validation;
+
+// Sourced from GDS repository
+// https://github.com/alphagov/notifications-utils/blob/main/tests/recipient_validation/test_email_address.py
+public static class EmailAddressValidator
+{
+    private static readonly string ValidLocalChars = @"a-zA-Z0-9.!#$%&'*+/=?^_`{|}~\-";
+    private static readonly string EmailRegexPattern = $@"^[{ValidLocalChars}]+@([^.@][^@\s]+)$";
+    private static readonly Regex HostnamePartRegex = new(@"^(xn|[a-z0-9]+)(-?-[a-z0-9]+)*$", RegexOptions.IgnoreCase);
+    private static readonly Regex TldPartRegex = new(@"^([a-z]{2,63}|xn--([a-z0-9]+-)*[a-z0-9]+)$", RegexOptions.IgnoreCase);
+
+    public static bool IsValid(string emailAddress)
+    {
+        emailAddress = StripAndRemoveObscureWhitespace(emailAddress);
```

Comment:
If I understand this correctly, we strip some characters for validation. Does it mean we then persist the email with those characters included?

Reply:
Sanitising the value before validation has been taken out of the validation process and applied separately with a ModelBinder. This way it has become reusable and when modified to be applicable on the Page model level, we could easily apply it solution-wide to improve security (as fallback for eliminating XSS, SQL injections etc. if any sneaked in due to not using MS HTML extensions).